components/director/internal/domain/application/service.go

```diff
@@ -301,6 +305,21 @@ func (s *service) Delete(ctx context.Context, id string) error {
 		return errors.Wrapf(err, "while loading tenant from context")
 	}
 
+	scenarios, runtimes, err := s.getScenariosAndRuntimes(ctx, appTenant, id)
+	if err != nil {
+		return errors.Wrapf(err, "while checking if application is in formation and runtime")
+	}
+
+	if len(scenarios) > 0 {
```

Comment:
this is abit messy in terms of error handling but when you split the methods it should become more straight forward out of the box